### ISSUE TITLE
Windows/Mac differences in presentation of some HTML-AAM implicit roles

### ISSUE THREAD

**Issue by cookiecrook** [w3c/html-aam#467]
While working on [HTML-AAM automated role tests](https://github.com/web-platform-tests/wpt/pull/39427), I noticed there were a few implicit role expectations that seemed to align with a Windows-style rendering, but did not align with the macOS renderings.

In particular, this test of a select list without a size attribute:
```html
<select data-testname="el-select-combobox" data-expectedrole="combobox" class="ex"><option>a1</option><option>a2</option></select>
```
Yields: `assert_equals: expected "combobox" but got "button"` in WebKit on macOS.

A true `combobox` is always (or almost always?) rendered as a text field with text-carat insertion and an associated menu. That's not the case for simple `<select>` menus on Mac. They actually render a button that triggers a menu. 

#### Screen shots of simple select (not a "combobox"):
<img width="54" alt="collapsed popup button" src="https://user-images.githubusercontent.com/75867/230691035-c642d342-e4cc-45af-b5e8-248ee1d520cc.png"> <img width="67" alt="expanded popup menu" src="https://user-images.githubusercontent.com/75867/230691161-602d0aef-b27b-4b09-a5d6-746a81e4f985.png">

I think this represents a platform-level user interface different that should be conveyed to the user of assistive technology. A Mac VoiceOver user would expect they could literally type or select text in a combobox, and that's not a valid assumption here.

In a related but inverse scenario, macOS renders `<input type="number>` as a text field with enclosed steppers, as opposed to the Windows-style "spin button."

#### Screen shot of simple number input (not a "button"):
<img width="161" alt="text insertion field with 'stepper' buttons" src="https://user-images.githubusercontent.com/75867/230691888-2957dfd6-2301-427c-a902-ffe597e06647.png">

Note: I'll try to add some screen shots of the equivalent Windows control to further the discussion.

### In summary
- It's possible some of the HTML-AAM implicit roles need changes, or platform-specific allowances. (This HTML-AAM issue.)
- It's seems clear to me that some presentation-specific role names (spin **_button_**, progress **_bar_**) in ARIA can result in UI confusion. It may be worth considering replacing these with less UI-centric synonyms, perhaps `progressbar -> progress` and `spinbutton -> number` or `spinbutton -> stepper`. (New ARIA issue?)

**Comment by JAWS-test** [comment]
The difference in select is indeed relevant, but in spinbutton (input type=number) I don't see the difference between Mac and Windows, m.E. in both the HTML element is implemented the same. That the ARIA role spinbutton allows additional variants is correct, but has nothing to do with the operating systems (see https://github.com/w3c/aria-practices/issues/1155)

**Comment by JAWS-test** [comment]
Related: https://github.com/w3c/core-aam/issues/111

**Comment by spectranaut** [comment]
Discussed in the ARIA working group today: https://www.w3.org/2023/04/20-aria-minutes#t05

**Comment by scottaohara** [comment]
Reminder to consider the points raised in this issue when working on this one: https://github.com/w3c/html-aam/issues/473

**Comment by smhigley** [comment]
Another consideration that came up in the call -- certain attributes like `aria-invalid` are valid on a `<select>`, but not on a `button` role.

**Comment by JAWS-test** [comment]
> Another consideration that came up in the call -- certain attributes like aria-invalid are valid on a select, but not on a button role.

Also true for
* aria-errormessage
* aria-readonly 
* aria-required

Furthermore, a select has a value and a label, a button has only a label, no value.

**Comment by spectranaut** [comment]
To schedule after TPAC
